diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/chapter3/files/NewlinesRule.kt

```diff
@@ -235,6 +247,48 @@ class NewlinesRule(private val configRules: List<RulesConfig>) : Rule("newlines"
         }
     }
 
+    private fun lengthFunListPar(node: ASTNode) {
+        val sizeParamList = node.findAllNodesWithSpecificType(VALUE_PARAMETER).size
+        if (!node.text.contains("\n") && (sizeParamList > configuration.maxParametersInOneLine || (node.text.length - 2) > configuration.maxLengthOneLine)
+        ) {
+            handleValueParameterList(node)
+        }
+    }
+
+    /**
+     * Check that newline is not placed before or after a colon
+     */
+    private fun handleColon(node: ASTNode) {
+        node.visit { colonNode ->
+            if (colonNode.elementType == COLON) {
+                val prevNewLine = colonNode
+                    .parent({ it.treePrev != null }, strict = false)
+                    ?.treePrev
+                    ?.takeIf {
+                        it.elementType == WHITE_SPACE && it.text.contains("\n")
+                    }
+                prevNewLine?.let {
+                    WRONG_NEWLINES.warnAndFix(configRules, emitWarn, isFixMode, "newline shouldn't be placed before colon", colonNode.startOffset, colonNode) {
+                        it.treeParent.removeChild(it)
+                    }
+                }
+                val nextNewLine = colonNode
+                    .parent({ it.treeNext != null }, strict = false)
+                    ?.treeNext
+                    ?.takeIf {
+                        it.elementType == WHITE_SPACE && it.text.contains("\n")
+                    }
+                nextNewLine?.let {
+                    WRONG_NEWLINES.warnAndFix(configRules, emitWarn, isFixMode, "newline shouldn't be placed after colon", colonNode.startOffset, colonNode) {
```

Comment:
I think it should be allowed, because it may be the only way to split a long line:
```kotlin
fun foo(a: Int,
    veryLongArgumentName:
        VeryLongTypeName)
```